app/services/api/v1/response_flow_from_email.rb

```diff
@@ -0,0 +1,37 @@
+class Api::V1::ResponseFlowFromEmail
+  attr_reader :params, :user, :time_period_id, :emotion_id
+
+  def initialize(params, user)
+    @params = params
+    @user = user
+    @time_period_id = params[:time_period_id]
+    @emotion_id = params[:emotion_id]
+  end
+
+  def call
+    set_existed_response
+    if @existed_response.present?
+      update_response
+      { success: true, response: @existed_response }
+    else
+      create_response
+      { success: true, response: @new_response }
+    end
+  rescue StandardError => e
+    { success: false, error: e }
+  end
+
+  private
+
+  def create_response
+    @new_response = user.responses.create!(time_period_id: time_period_id, emotion_id: emotion_id, step: 'MemeSelection')
+  end
+
+  def update_response
+    @existed_response.update!(emotion_id: emotion_id, not_working: false, step: 'MemeSelection')
+  end
+
+  def set_existed_response
```

Comment:
Maybe just use memoization here?
```ruby
def existing_response
   @existing_response ||= ...
end
```
and then instead of having the other methods use the global variable they can just call by the method name. For example the call method would then be just:
```ruby
def call
   if existing_response.present?
     ...
```